# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1261035513 18000
#      Thu Dec 17 02:38:33 2009 -0500
# Node ID fdc3a43c0be84b632f16a135a29ae6ec0f881c1e
# Parent  b22a2f4b34aadfa6fe0f40e9e44e668378d237e7
avoid conflict with gnulib defining glob to be rpl_glob

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,14 @@
+2009-12-17  John W. Eaton  <jwe@octave.org>
+
+	* glob-match.h, glob-match.cc (glob_match::glob_internal):
+	Rename from glob_match::glob and make private.
+	* glob-match.h (glob_match::glob): Forward to glob_internal.
+
 2009-12-15  Jaroslav Hajek  <highegg@gmail.com>
 
 	* Array.cc (Array<T>::index (const idx_vector&)): Fix determining of
 	result dimensions.
 
 2009-12-15  Jaroslav Hajek  <highegg@gmail.com>
 
 	* Range.cc (Range::elem, Range::checkelem, Range::index):
diff --git a/liboctave/glob-match.cc b/liboctave/glob-match.cc
--- a/liboctave/glob-match.cc
+++ b/liboctave/glob-match.cc
@@ -76,17 +76,17 @@ static bool
 single_match_exists (const std::string& file)
 {
   file_stat s (file);
 
   return s.exists ();
 }
 
 string_vector
-glob_match::glob (void)
+glob_match::glob_internal (void)
 {
   string_vector retval;
 
   int npat = pat.length ();
 
   int k = 0;
 
   for (int i = 0; i < npat; i++)
diff --git a/liboctave/glob-match.h b/liboctave/glob-match.h
--- a/liboctave/glob-match.h
+++ b/liboctave/glob-match.h
@@ -66,25 +66,30 @@ public:
   void set_pattern (const std::string& p) { pat = p; }
 
   void set_pattern (const string_vector& p) { pat = p; }
 
   bool match (const std::string&);
 
   Array<bool> match (const string_vector&);
 
-  string_vector glob (void);
+  // We forward to glob_internal here to avoid problems with gnulib's
+  // glob.h defining glob to be rpl_glob.
+
+  string_vector glob (void) { return glob_internal (); }
 
 private:
 
   // Globbing pattern(s).
   string_vector pat;
 
   // Option flags.
   unsigned int flags;
+
+  string_vector glob_internal (void);
 };
 
 #endif
 
 /*
 ;;; Local Variables: ***
 ;;; mode: C++ ***
 ;;; End: ***
